lib/orderbook/MatchesProcessor.ts

```diff
@@ -39,33 +39,35 @@ class MatchesProcessor {
       let makerCurrency: string;
       let takerAmount: number;
       let makerAmount: number;
+       // TODO: use configurable amount of subunits/satoshis per token for each currency
       if (taker.quantity > 0) {
         // we are buying the base currency
         takerCurrency = baseCurrency;
         makerCurrency = quoteCurrency;
-        takerAmount = taker.quantity;
-        makerAmount = taker.quantity * maker.price;
+        takerAmount = taker.quantity * 100000000;
+        makerAmount = taker.quantity * maker.price * 100000000;
       } else {
         // we are selling the base currency
         takerCurrency = quoteCurrency;
         makerCurrency = baseCurrency;
-        takerAmount = taker.quantity * maker.price;
-        makerAmount = taker.quantity;
+        takerAmount = taker.quantity * maker.price * -100000000;
+        makerAmount = taker.quantity * -100000000;
```

Comment:
Not sure why there is a need for negative numbers.

Reply:
Because `taker.quantity` is negative in this `else` block, the `if` statement above checks `if (taker.quantity > 0)`. This indicates that the taker order (aka the local order) is a sell order. Multiplying by a negative number brings it back to positive.